### PR TITLE
MULE-18580: ZonedDateTime Parameters are not parsed from Strings

### PR DESCRIPTION
**Pull request by dabreu**

### CODE REVIEW

File: tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java [hunk 1]
```diff
@@ -7,22 +7,37 @@
 
 package org.mule.test.some.extension;
 
+import org.mule.runtime.extension.api.annotation.Configuration;
+import org.mule.runtime.extension.api.annotation.Configurations;
```
Comment (ndinu) [inline tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java:11]: remove unused import

File: tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java [hunk 2]
```diff
@@ -7,22 +7,37 @@
...
 import org.mule.runtime.extension.api.annotation.dsl.xml.Xml;
 import org.mule.runtime.extension.api.annotation.error.ErrorTypes;
+import org.mule.sdk.api.annotation.param.Optional;
+import org.mule.sdk.api.annotation.param.Parameter;
 import org.mule.test.heisenberg.extension.HeisenbergErrors;
 
+import java.time.ZonedDateTime;
+
 @Extension(name = "SomeExtension")
+@Configuration(name = "config")
```
Comment (SebaElizalde) [inline tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java:26]: why is this needed?

File: tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeOps.java [hunk 3]
```diff
@@ -26,8 +29,27 @@
...
 
+  public SomeExtension retrieveConfiguration(@Config SomeExtension config) {
+    return config;
+  }
+
+  public ZonedDateTime retrieveOperationZonedDateTime() {
+    return zonedDateTime;
+  }
+
+  public boolean isAfter(ZonedDateTime zonedDateTimeParam, @Config SomeExtension config) {
```
Comment (SebaElizalde) [inline tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeOps.java:46]: it is not clear to me the purpose of this operation it just tests the isAfter method of the Java class

**Review by SebaElizalde — CHANGES_REQUESTED** [review]

File: tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java (hunk 1 above)
Comment (dabreu) [reply tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java:11]: will do

File: tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java (hunk 2 above)
Comment (dabreu) [reply tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java:26]: I added the name in order to reference the config on the parameter-value-resolver.xml file

File: tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeOps.java (hunk 3 above)
Comment (dabreu) [reply tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeOps.java:46]: The intention of this method was to add a test scenario to validate that a ZonedDateTime operation parameter/argument is being correctly parsed from the configured string on the flow xml file.

Comment (SebaElizalde) [reply tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeOps.java:46]: Maybe make the test just return the parameter then? Plus adding a parameter to the SomeOps class like you did:
```
  @Parameter
  @Optional
  private ZonedDateTime zonedDateTime;
```
 makes all methods/operations defined in that class also have that parameter already. So maybe this can be simplified so there are less parameters and it is not so confusing.

File: tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java (hunk 2 above)
Comment (SebaElizalde) [reply tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java:25]: oh, it became required because you added an optional parameter to the config?

File: tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java [hunk 4]
```diff
@@ -7,22 +7,36 @@
...
 @Export(classes = CustomConnectionException.class)
-@Sources({SomeEmittingSource.class, AnotherEmittingSource.class, YetAnotherEmittingSource.class})
+@Sources({SomeEmittingSource.class, AnotherEmittingSource.class, YetAnotherEmittingSource.class,
+    ParameterEmittingSource.class})
 @Xml(namespace = "http://www.mulesoft.org/schema/mule/some", prefix = "some")
 public class SomeExtension {
 
+  @Parameter
+  @Optional
+  private ZonedDateTime zonedDateTime;
```
Comment (SebaElizalde) [inline tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeExtension.java:37]: Also if there aren't any differences in picking up a value from a config or a source or a parameter, maybe it is not needed to test all possible combinations (config + source + operation + another operation), but it does not hurt either... so I leave that up to you.

File: tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeOps.java (hunk 3 above)
Comment (dabreu) [reply tests/test-extensions/some-extension/src/main/java/org/mule/test/some/extension/SomeOps.java:46]: Yes, makes sense, I will simplify the operation as you suggested

**Review by SebaElizalde — APPROVED** [review]
